middleware/mongoose.ts

```diff
@@ -0,0 +1,16 @@
+import mongoose from "mongoose";
+import type { NextApiRequest, NextApiResponse } from "next";
+
+export default async (
+  _req: NextApiRequest,
+  _res: NextApiResponse,
+  next: () => void
+) => {
+  if (mongoose.connections[0].readyState) {
+    // already connected
```

Comment:
ease of understanding: great commenting to ensure people who are not familiar to mongoose can understand easily.